### PR TITLE
feat: add support for OpenAPI version 3.1.0

### LINKED ISSUE #1749: Add OpenAPI 3.1 support
#### Reason/Context
Right now we already have support for [OpenAPI 3.0](https://github.com/asyncapi/modelina/blob/master/src/processors/OpenAPIInputProcessor.ts) and [Swagger 2.0](https://github.com/asyncapi/modelina/blob/master/src/processors/SwaggerInputProcessor.ts). But we should also support the newest 3.1 version: https://github.com/OAI/OpenAPI-Specification/blob/main/versions/3.1.0.md

To support it we need to alter the existing OpenAPI input processor to also support the new 3.1 version: https://github.com/asyncapi/modelina/blob/master/src/processors/OpenAPIInputProcessor.ts

This does 

You can use the following OpenAPI 3.1 input document for testing (or come up with your own):
```json
{
  "openapi": "3.1.0",
  "info": {
    "version": "1.0.0",
    "title": "Swagger Petstore",
    "license": {
      "name": "MIT"
    }
  },
  "servers": [
    {
      "url": "http://petstore.swagger.io/v1"
    }
  ],
  "paths": {
    "/pets": {
      "get": {
        "summary": "List all pets",
        "operationId": "listPets",
        "tags": [
          "pets"
        ],
        "parameters": [
          {
            "name": "limit",
            "in": "query",
            "description": "How many items to return at one time (max 100)",
            "required": false,
            "schema": {
              "type": "integer",
              "maximum": 100,
              "format": "int32"
            }
          }
        ],
        "responses": {
          "200": {
            "description": "A paged array of pets",
            "headers": {
              "x-next": {
                "description": "A link to the next page of responses",
                "schema": {
                  "type": "string"
                }
              }
            },
            "content": {
              "application/json": {
                "schema": {
                  "$ref": "#/components/schemas/Pets"
                }
              }
            }
          },
          "default": {
            "description": "unexpected error",
            "content": {
              "application/json": {
                "schema": {
                  "$ref": "#/components/schemas/Error"
                }
              }
            }
          }
        }
      },
      "post": {
        "summary": "Create a pet",
        "operationId": "createPets",
        "tags": [
          "pets"
        ],
        "requestBody": {
          "content": {
            "application/json": {
              "schema": {
                "$ref": "#/components/schemas/Pet"
              }
            }
          },
          "required": true
        },
        "responses": {
          "201": {
            "description": "Null response"
          },
          "default": {
            "description": "unexpected error",
            "content": {
              "application/json": {
                "schema": {
                  "$ref": "#/components/schemas/Error"
                }
              }
            }
          }
        }
      }
    },
    "/pets/{petId}": {
      "get": {
        "summary": "Info for a specific pet",
        "operationId": "showPetById",
        "tags": [
          "pets"
        ],
        "parameters": [
          {
            "name": "petId",
            "in": "path",
            "required": true,
            "description": "The id of the pet to retrieve",
            "schema": {
              "type": "string"
            }
          }
        ],
        "responses": {
          "200": {
            "description": "Expected response to a valid request",
            "content": {
              "application/json": {
                "schema": {
                  "$ref": "#/components/schemas/Pet"
                }
              }
            }
          },
          "default": {
            "description": "unexpected error",
            "content": {
              "application/json": {
                "schema": {
                  "$ref": "#/components/schemas/Error"
                }
              }
            }
          }
        }
      }
    }
  },
  "components": {
    "schemas": {
      "Pet": {
        "type": "object",
        "required": [
          "id",
          "name"
        ],
        "properties": {
          "id": {
            "type": "integer",
            "format": "int64"
          },
          "name": {
            "type": "string"
          },
          "tag": {
            "type": "string"
          }
        }
      },
      "Pets": {
        "type": "array",
        "maxItems": 100,
        "items": {
          "$ref": "#/components/schemas/Pet"
        }
      },
      "Error": {
        "type": "object",
        "required": [
          "code",
          "message"
        ],
        "properties": {
          "code": {
            "type": "integer",
            "format": "int32"
          },
          "message": {
            "type": "string"
          }
        }
      }
    }
  }
}
```

Happy to help further if you get stuck on something :v:

### PR DESCRIPTION
**Pull request by Athul0491**
## Description
Adds support for OpenAPI version 3.1

## Related Issue
Fixes #1749

## Checklist
- [x] The code follows the project's coding standards and is properly linted (`npm run lint`).
- [x] Tests have been added or updated to cover the changes.
- [x] Documentation has been updated to reflect the changes.
- [x] All tests pass successfully locally.(`npm run test`).

### CODE REVIEW

**Review by jonaslagoni — CHANGES_REQUESTED** [review]
@Athul0491 please add an example and adapt the docs to reflect the new supported input✌️

Let me know if you need pointers

**Comment by Athul0491** [comment]
@jonaslagoni Will an example similar to [openapi-from-object](https://github.com/asyncapi/modelina/tree/master/examples/openapi-from-object) be sufficient?

**Comment by jonaslagoni** [comment]
@Athul0491 yep exactly 👍 also make sure you adapt test file for the input processor

**Comment by jonaslagoni** [comment]
@Athul0491 have you had time to take another look at it? :v:

File: examples/openapi-v3_1-from-object/index.spec.ts [hunk 1]
```diff
@@ -0,0 +1,15 @@
+const spy = jest.spyOn(global.console, 'log').mockImplementation(() => {
+  return;
+});
+import { generate } from './index';
```
Comment (jonaslagoni) [inline examples/openapi-v3_1-from-object/index.spec.ts:4]: Please commit the snapshots as well 🙂

**Review by jonaslagoni — CHANGES_REQUESTED** [review]
I am pretty sure the package-lock changes are stopping the tests for working. I have seen that error before 🤔 

Did you make the `npm i` with node 18 or 16? 🤔 

Everything else looks fine to me tbh 👍

**Comment by Athul0491** [comment]
I ran `npm i` with node 18.18.2

**Comment by jonaslagoni** [comment]
@Athul0491 do you mind grabbing the package-lock from master and copy it over, and do a simple `npm i` again? 🤔

**Review by jonaslagoni — APPROVED** [review]
Perfect 👌

**Comment by jonaslagoni** [comment]
@all-contributors please add @Athul0491 for code, example, docs, test

**Pull request merged**